src/domain/opendatafabric/src/identity/dataset_refs.rs

```diff
@@ -739,79 +739,117 @@ impl std::cmp::PartialOrd for DatasetRefAny {
     }
 }
 
+////////////////////////////////////////////////////////////////////////////////
+// DatasetRefPattern
+////////////////////////////////////////////////////////////////////////////////
+
 #[derive(Debug, Clone, Eq, PartialEq)]
-pub enum DatasetRefPattern {
+pub enum DatasetRefPatternLocal {
     Ref(DatasetRef),
-    Pattern(Option<AccountName>, DatasetNamePattern),
+    Pattern(DatasetPattern),
 }
 
-impl DatasetRefPattern {
-    pub fn is_match(&self, dataset_hande: &DatasetHandle) -> bool {
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub enum DatasetRefPatternAny {
```

Comment:
I think this should be called `DatasetRefAnyPattern` as it's a superset of the `DatasetRefAny` type.